cupyx/scipy/sparse/_data.py

```diff
@@ -227,25 +225,29 @@ def _arg_min_or_max(self, axis, out, op, compare):
             if self.nnz == 0:
                 return 0
             else:
-                zero = self.dtype.type(0)
+                zero = cupy.asarray(self.dtype.type(0))
                 mat = self.tocoo()
 
                 mat.sum_duplicates()
 
                 am = op(mat.data)
                 m = mat.data[am]
 
-                if compare(m, zero):
+                zero_cmp = cupy.where(compare(m, zero), True, False)
+
+                if zero_cmp:
```

Comment:
Is this accurate?

Reply:
The element of `zero_cmp` is stored on GPU memory, and it is transferred to CPU memory for Python code path branching here. My suggestion is to use `cupy.where` to branch a code path in CUDA!